mongo/datadog_checks/mongo/api.py

```diff
@@ -49,49 +36,36 @@ def server_info(self, session=None):
     def list_database_names(self, session=None):
         return self._cli.list_database_names(session)
 
-    def _initialize(self):
+    def _initialize(self, replicaset=None):
         self._log.debug("Connecting to '%s'", self._config.hosts)
 
         is_master_payload = self['admin'].command('isMaster')
         is_arbiter = is_master_payload.get('arbiterOnly', False)
 
+        self._log.debug("is_master_payload: %s", is_master_payload)
         if not is_arbiter and self._config.do_auth:
             self._log.info("Using '%s' as the authentication database", self._config.auth_source)
-            self._authenticate()
+            options = {
+                'host': self._config.server if self._config.server else self._config.hosts,
+                'username': self._config.username,
+                'password': self._config.password,
```

Comment:
Does this work correctly if no username or passwords are provided? (Is the CLI ok with receiving those as None?)